File: app/src/main/java/demo/victormunoz/githubusers/di/module/UserDetailsPresenterModule.java
package demo.victormunoz.githubusers.di.module;

import android.app.Activity;
import android.support.annotation.NonNull;

import dagger.Module;
import dagger.Provides;
import demo.victormunoz.githubusers.di.scope.ActivityScope;
import demo.victormunoz.githubusers.features.userDetails.UserDetailsContract.PresenterListener;
import demo.victormunoz.githubusers.features.userDetails.UserDetailsContract.UserActionsListener;
import demo.victormunoz.githubusers.features.userDetails.UserDetailsPresenter;
import demo.victormunoz.githubusers.model.network.GithubService;

@Module(includes = {GitHubModule.class, PicassoModule.class, CircleTransformationModule.class})
public class UserDetailsPresenterModule {
    @NonNull
    private final PresenterListener mUsersPresenterListener;


    public UserDetailsPresenterModule(@NonNull Activity activity){
        mUsersPresenterListener = (PresenterListener) activity;
    }

    @Provides
    @ActivityScope
    @NonNull
    UserActionsListener providesUserPresenterInterface(GithubService githubService){
        return new UserDetailsPresenter(mUsersPresenterListener, githubService);
    }


}

File: app/src/main/java/demo/victormunoz/githubusers/di/module/UserDetailsPresenterModule.kt
package demo.victormunoz.githubusers.di.module

import android.app.Activity
import com.trello.rxlifecycle2.components.support.RxAppCompatActivity

import dagger.Module
import dagger.Provides
import demo.victormunoz.githubusers.di.scope.ActivityScope
import demo.victormunoz.githubusers.features.userDetails.UserDetailsContract.ViewListener
import demo.victormunoz.githubusers.features.userDetails.UserDetailsContract.PresenterListener
import demo.victormunoz.githubusers.features.userDetails.UserDetailsPresenter
import demo.victormunoz.githubusers.network.api.ApiService
import demo.victormunoz.githubusers.network.image.ImageService

@Module(includes = [(GitHubModule::class), (PicassoModule::class)])
class UserDetailsPresenterModule(activity: Activity) {

    private var viewListener = activity as ViewListener
    private var lifecycle = (activity as RxAppCompatActivity).lifecycle()

    @Provides
    @ActivityScope
    internal fun providesUserPresenterInterface(githubService: ApiService, imageService: ImageService): PresenterListener {
        return UserDetailsPresenter(viewListener, githubService, imageService, lifecycle)
    }

}
